EpiAware/src/latent-processes.jl

```diff
@@ -1,32 +1,25 @@
-const STANDARD_RW_PRIORS =
-    (var_RW_dist = truncated(Normal(0.0, 0.05), 0.0, Inf), init_rw_value_dist = Normal())
-
-
-"""
-    random_walk(n, ϵ_t = missing; latent_process_priors = (var_RW_dist = truncated(Normal(0., 0.05), 0., Inf),), ::Type{T} = Float64) where {T <: Real}
-
-Constructs a random walk model.
-
-# Arguments
-- `n`: The number of time steps.
-- `ϵ_t`: The random noise vector. Defaults to `missing`, in which case it is sampled from the standard multivariate normal distribution.
-- `latent_process_priors`: The prior distribution for the latent process parameters. Defaults to `(var_RW_dist = truncated(Normal(0., 0.05), 0., Inf),)`.
+function default_rw_priors()
+    return (
+        var_RW_dist = truncated(Normal(0.0, 0.05), 0.0, Inf),
+        init_rw_value_dist = Normal(),
+    )
+end
 
-# Returns
-- `rw`: The random walk process.
-- `σ_RW`: The standard deviation of the random walk process.
-"""
 @model function random_walk(
     n,
     ϵ_t = missing,
     ::Type{T} = Float64;
-    latent_process_priors = STANDARD_RW_PRIORS,
-) where {T<:Real}
+    latent_process_priors = default_rw_priors(),
+) where {T<:AbstractFloat}
     rw = Vector{T}(undef, n)
     ϵ_t ~ MvNormal(ones(n))
     σ²_RW ~ latent_process_priors.var_RW_dist
     init_rw_value ~ latent_process_priors.init_rw_value_dist
     σ_RW = sqrt(σ²_RW)
-    rw .= init_rw_value .+ cumsum(σ_RW * ϵ_t)
-    return rw, (; σ_RW, init_rw_value)
+
+    rw[1] = init_rw_value + σ_RW * ϵ_t[1]
```

Comment:
why the change here?

Reply:
I was declaring `rw` into memory, _then_ declaring another RHS vector and assigning. I wasn't sure if the compiler was abstracting that away (got to start some proper profiling...), but this way it definitely isn't.